modules/python/python-tf.c

```diff
@@ -119,10 +121,14 @@ _py_convert_return_value_to_result(PythonTfState *state, const gchar *function_n
                 evt_tag_str("exception", _py_format_exception_text(buf, sizeof(buf))));
       _py_finish_exception_handling();
 
+      scratch_buffers_reclaim_marked(marker);
       Py_DECREF(ret);
       return FALSE;
     }
 
+  g_string_append(result, value->str);
```

Comment:
Although the naming was misleading, I liked that the `python-types` functions appended to the `GString`s, exactly because of these use cases, where we now need to do copies, which could have been optimized with the appending logic.

As an alternative I would propose to:
  1. Rename the functions which append, to clearly communicate the behavior.
  2. Fix the datetime conversion to append.
  3. Modify the unit tests, to start with a non-empty `GString`, so the appending behavior is unit tested.